src/main/java/carracing/step4/Cars.java

```diff
@@ -0,0 +1,55 @@
+package carracing.step4;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static java.util.stream.Collectors.toList;
+
+public class Cars {
+    private final List<Car> cars;
+    private final Supplier<Integer> numberGenerator;
+
+    public Cars(List<String> carNames, Supplier<Integer> numberGenerator) {
+        this.cars = new ArrayList<>(carNames.size());
+        for (String name : carNames) {
+            this.cars.add(new Car(name));
+        }
+        this.numberGenerator = numberGenerator;
+    }
+
+    public void moveAllCars() {
+        for (int i = 0; i < cars.size(); i++) {
+            moveCar(i);
+        }
+    }
+
+    private void moveCar(int carNum) {
+        cars.get(carNum).move(numberGenerator.get());
+    }
+
+    public List<CarVo> getAllCarsTrace() {
+        return cars.stream().map(car -> new CarVo(car.getName(), car.getMoveCnt())).collect(toList());
```

Comment:
```suggestion
        return cars.stream()
                .map(car -> new CarVo(car.getName(), car.getMoveCnt()))
                .collect(toList());
```
줄바꿈을 통해 가독성을 높일 수 있습니다ㅎㅎ
보통 메서드 체이닝에선 한 라인에는 하나의 점을 찍는 편입니다